### PR TITLE
[PSDK-33] Define list of errors

### PR DESCRIPTION
**Pull request by yuga-cb**
### What changed? Why?
Define the list of API errors

### CODE REVIEW

File: lib/coinbase/errors.rb [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+      message = "API Error \n"
+      message += "HTTP status code: #{http_code}\n" if http_code
+      message += "API error code: #{api_code}\n" if api_code
+      message += "API error message: #{api_message}\n" if api_message
+      message
+    end
+  end
+
+  class UnimplementedError < APIError
+  end
```
Comment (alex-stone) [inline lib/coinbase/errors.rb:97]: **nit**: Pretty standard to inline these so we can fit all of these on a single line:
```suggestion
  class UnimplementedError < APIError; end
```

File: lib/coinbase/errors.rb [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    # @param err [Coinbase::Client::APIError] The underlying error object.
+    # @return [APIError] The specific APIError object.
+    def self.from_error(err)
+      raise ArgumentError, 'Argument must be a Coinbase::Client::APIError' unless err.is_a? Coinbase::Client::ApiError
+      return APIError.new(err) unless err.response_body
+
+      body = JSON.parse(err.response_body)
+      api_code = body['code']
+
+      case api_code
```
Comment (alex-stone) [inline lib/coinbase/errors.rb:34]:
```suggestion
      case body['code']
```

File: lib/coinbase/errors.rb (hunk 1 above)
Comment (yuga-cb) [reply lib/coinbase/errors.rb]: done

File: lib/coinbase/errors.rb [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+module Coinbase
+  # A wrapper for API errors to provide more context.
+  class APIError < StandardError
+    attr_reader :api_code, :api_message
+
+    # Initializes a new APIError object.
+    # @param err [Coinbase::Client::APIError] The underlying error object.
+    def initialize(err)
+      super
+      @e = err
```
[the remark this answers is not in the document]
Comment (yuga-cb) [reply lib/coinbase/errors.rb]: done

File: lib/coinbase/errors.rb (hunk 2 above)
Comment (yuga-cb) [reply lib/coinbase/errors.rb]: done

**Review by alex-stone — APPROVED** [review]

**Pull request merged**